### PR TITLE
Improve design document

### PR DESCRIPTION
**Pull request by rohrerj**

### CODE REVIEW

File: doc/BorderRouter.rst [hunk 1]
```diff
@@ -1,182 +1,220 @@
...
+Abstract
+===========
 
 Right now, the performance of the border router is very much limited because a single goroutine per
 border router interface is responsible for reading, parsing, processing and forwarding the packets.
+
+Background
+===========
+
 Redesigning this pipeline is expected to lead to significant performance improvements.
```
Comment (mawyss) [inline doc/BorderRouter.rst]: - Move first sentence at the end of this paragraph. ("Redesigning the current router pipeline is...")
- Start with "The current border router design..."

File: doc/BorderRouter.rst [hunk 2]
```diff
@@ -1,182 +1,220 @@
...
 Previously, there also existed a design which did something similar than the proposed design in
 this document.
-[See](https://github.com/scionproto/scion/tree/92531f5cb62197b9d705001c13e5a6bdb7ba1fa4/go/border).
+`See <https://github.com/scionproto/scion/tree/92531f5cb62197b9d705001c13e5a6bdb7ba1fa4/go/border>`_.
 
-## Overview
+Proposal
+========
 
 The pipeline gets changed to have seperate goroutines for the receiving, processing and forwarding steps.
```
Comment (mawyss) [inline doc/BorderRouter.rst:27]: Also mention that we don't need to allocate memory at runtime anymore.
It is all preallocated

File: doc/BorderRouter.rst [hunk 3]
```diff
@@ -1,182 +1,220 @@
...
+Pool size
+^^^^^^^^^^^
 
-Since a pool of packet buffers is bound to a receiver, the size of the pool can be configured for each
-receiver seperately.
-It makes sense to make this configurable on a per border router interface level because not every
-interface might have the same expected load and hence not every interface requires the same pool size.
+The size of the packet buffer pool of the receivers can be configured.
 An optimal value should be derivable from the maximum latency of processing and forwarding a packet,
 the speed of the network interfaces and the number of available CPU cores.
```
Comment (mawyss) [inline doc/BorderRouter.rst]: Maybe add the computation that Matthias did?

File: doc/BorderRouter.rst [hunk 4]
```diff
@@ -1,182 +1,220 @@
...
-### Multiple receivers per border router interface
+Multiple receivers per border router interface
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 We could deploy multiple packet receivers per border router interface and use eBPF to make sure that
 all packets that belong to the same flow are received by the same receiver.
 Because the rest remains unchanged we would still have the "no-reordering" guarantee and significantly
 increase the read speed.
 
-### Lock goroutines to threads
```
Comment (mawyss) [inline doc/BorderRouter.rst:133]: We was this completely removed?

Comment (rohrerj) [reply doc/BorderRouter.rst:133]: I re-added the first part of the sentence.

File: doc/BorderRouter.rst (hunk 3 above)
Comment (mawyss) [reply doc/BorderRouter.rst]: Then I guess we don't need the previous sentence anymore?
("should be derivable from the maximum latency...")

Also: an optimal could -> an optimal value could

**Pull request merged**